File: C++ ad y ad/3_ Orientado a objetos/S05.2_ Consolas/Solucion_consolas/Consolas/src/mando/mando.cpp
#include "mando.h"

#include <iostream>
#include <cstddef>

using namespace std;


Mando::Mando()
{
	consola = NULL;
}
Mando::~Mando()
{
	cout << "El mando fue liberado de la memoria." << endl << endl;
}


void Mando::cambiarVolumenConsola(int volumen)
{
	consola->setVolumen(volumen);
}
void Mando::cambiarBrilloConsola(int brillo)
{
	consola->setBrillo(brillo);
}

void Mando::conectarALaConsola(Consola* consola)
{
	this->consola = consola;
}
void Mando::desconectar()
{
	consola = NULL;
}

bool Mando::estaConectado()
{
	return consola != NULL;
}

Consola* Mando::getConsola()
{
	return consola;
}

File: C# ad y ad/2_ Orientado a objetos/S05.2_ Consolas/Solucion_consolas/Consolas/Src/Mando/Mando.cs

namespace Consolas
{
	class Mando
	{
		Consola consola;


		public Mando()
		{
			consola = null;
		}


		public void CambiarVolumenConsola(int volumen)
		{
			consola.SetVolumen(volumen);
		}
		public void CambiarBrilloConsola(int brillo)
		{
			consola.SetBrillo(brillo);
		}

		public void ConectarALaConsola(Consola consola)
		{
			this.consola = consola;
		}
		public void Desconectar()
		{
			consola = null;
		}

		public bool EstaConectado()
		{
			return consola != null;
		}

		public Consola GetConsola()
		{
			return consola;
		}
	}
}
